### PR TITLE
Buckling cleanup and documentation. Borg buckle time tweak.

### PR DESCRIPTION
**Pull request by Yenwodyah**
## About The Pull Request
 * Added / improved documentation for buckling procs and variables
 * Removed / moved some unused things (removed 'buckling' var on mob, moved can_unbuckle() and can_buckle() from mob to living, removed can_unbuckle() and can_buckle() from slimes because they were ignoring everywhere it was checked anyways)
 * Moved can_buckle() check to is_buckle_possible() with the rest of the checks
 * Allowed mobs to buckle other mobs to things on the same turf as them ( I don't see why this was blocked in the first place. We have mobs on the same turf as each other all the time)
* Changed silicons to use user_buckle_mob() instead of their own do_after system - now slightly longer to buckle mobs from another turf but instant to buckle mobs from the same turf. This means that borgs can't combatspin people who are still standing but have a slight slowdown, but can load people even faster if they're stunned/incapacitated and lying down. (But honestly, I did it for consistency, not balance)

## Why It's Good For The Game
Documentation good. Consistency good.

## Changelog
:cl:
tweak: Borgs take slightly longer to load people from adjacent tiles but can load people instantly from the same tile
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
> **but instant to buckle mobs from the same turf**

👀 👀 👀 

Finally, a way for borgs to capitalize on knockdowns.

**Comment by ATH1909** [comment]
So the old time needed to buckle someone to you as a borg was 0.5 seconds; what are the new times?

Also, how does this affect mulebots and tamed megacarp?

**Comment by Yenwodyah** [comment]
New times is just the same as buckling another person to anything else - 2 seconds adjacent, instant same tile.
No other mobs should be affected mechanically

File: code/game/objects/buckling.dm [hunk 1]
```diff
@@ -41,44 +46,50 @@
 	if(Adjacent(user) && can_buckle && has_buckled_mobs())
 		if(buckled_mobs.len > 1)
 			var/unbuckled = input(user, "Who do you wish to unbuckle?","Unbuckle Who?") as null|mob in sortNames(buckled_mobs)
-			if(user_unbuckle_mob(unbuckled,user))
-				return TRUE
+			return(user_unbuckle_mob(unbuckled,user))
```
Comment (Rohesie) [inline code/game/objects/buckling.dm:49]:
```suggestion
			return user_unbuckle_mob(unbuckled,user)
```

File: code/game/objects/buckling.dm [hunk 2]
```diff
@@ -41,44 +46,50 @@
...
 		else
-			if(user_unbuckle_mob(buckled_mobs[1],user))
-				return TRUE
+			return(user_unbuckle_mob(buckled_mobs[1],user))
```
Comment (Rohesie) [inline code/game/objects/buckling.dm:51]:
```suggestion
			return user_unbuckle_mob(buckled_mobs[1], user)
```

File: code/game/objects/buckling.dm [hunk 3]
```diff
@@ -41,44 +46,50 @@
...
+  *
+  * Arguments:
+  * M - The mob being buckled to src
+  * user - The mob buckling M to src
+  */
 /atom/movable/proc/mouse_buckle_handling(mob/living/M, mob/living/user)
 	if(can_buckle && istype(M) && istype(user))
-		if(user_buckle_mob(M, user))
-			return TRUE
+		return(user_buckle_mob(M, user))
```
Comment (Rohesie) [inline code/game/objects/buckling.dm:66]:
```suggestion
		return user_buckle_mob(M, user)
```

File: code/modules/mob/living/living.dm [hunk 4]
```diff
@@ -1675,6 +1675,13 @@
 			if(old_buckled.buckle_lying == 0 && (resting || HAS_TRAIT(src, TRAIT_FLOORED))) // The buckle forced us to stay up (like a chair)
 				set_lying_down() // We want to rest or are otherwise floored, so let's drop on the ground.
 
+///can the mob be buckled to something by default?
+/mob/proc/can_buckle()
+	return TRUE
+
+///can the mob be unbuckled from something by default?
+/mob/proc/can_unbuckle()
+	return TRUE
```
Comment (Rohesie) [inline code/modules/mob/living/living.dm:1684]: Could you move these procs into variables, or even better, a single bitflag variable? Something like `buckling_flags`?
Check the contribution guidelines on complex and redefined getters, for an example on how to better handle this kind of thing.

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 5]
```diff
@@ -1017,9 +1017,9 @@
 	cell = null
 
 /mob/living/silicon/robot/mouse_buckle_handling(mob/living/M, mob/living/user)
+	//Don't try buckling on INTENT_HARM so that silicons can search people's inventories without loading them
 	if(can_buckle && isliving(user) && isliving(M) && !(M in buckled_mobs) && ((user != src) || (a_intent != INTENT_HARM)))
-		if(buckle_mob(M))
-			return TRUE
+		return(user_buckle_mob(M,user, check_loc = FALSE))
```
Comment (Rohesie) [inline code/modules/mob/living/silicon/robot/robot.dm:1022]:
```suggestion
		return user_buckle_mob(M, user, check_loc = FALSE)
```

File: code/modules/mob/living/living.dm (hunk 4 above)
Comment (Yenwodyah) [reply code/modules/mob/living/living.dm:1684]: Sure

Comment (Yenwodyah) [reply code/modules/mob/living/living.dm:1684]: Oh man I didn't even notice but I forgot to change it to mob/living too lol

File: code/modules/mob/living/living_defines.dm [hunk 6]
```diff
@@ -154,3 +154,6 @@
 	var/icon/held_rh = 'icons/mob/pets_held_rh.dmi'
 	var/icon/head_icon = 'icons/mob/pets_held.dmi'//what it looks like on your head
 	var/held_state = ""//icon state for the above
+
+	/// Is this mob allowed to be buckled/unbuckled to/from things?
+	var/can_buckle_to = TRUE
```
Comment (Rohesie) [inline code/modules/mob/living/living_defines.dm:159]: Maybe `can_be_buckled`?

**Pull request merged**